manifests/prepare/package.pp

```diff
@@ -28,12 +38,41 @@
     $mode = '0644'
   }
 
-  file { $local_package_file_path:
-    ensure  => file,
-    owner   => $puppet_agent::params::user,
-    group   => $puppet_agent::params::group,
-    mode    => $mode,
-    source  => $source,
-    require => File[$puppet_agent::params::local_packages_dir],
+  # REMIND: redhat/suse with absolute_source
+  # REMIND: debian with absolute_source
+  # REMIND: solaris 10
+  # REMIND: solaris 11 with manage_repo
+  # REMIND: aix
+  # REMIND: darwin
+  # REMIND: suse 11 and PE
+  if $puppet_agent::collection =~ /core/ and $facts['os']['family'] =~ /windows/ {
```

Comment:
```suggestion
  if $puppet_agent::collection and $puppet_agent::collection =~ /core/ and $facts['os']['family'] =~ /windows/ {
```
In my MacOS work ran into an issue here when a collection wasn't set at all, this fixed that.

Reply:
I had the same issue in debian, redhat, suse, so I added a commit to fix all of them.